substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/image/NativeImageDebugInfoProvider.java

```diff
@@ -958,11 +976,33 @@ public int getFrameSize() {
 
         @Override
         public boolean isDeoptTarget() {
-            return methodName().endsWith(HostedMethod.METHOD_NAME_DEOPT_SUFFIX);
+            return name().endsWith(HostedMethod.METHOD_NAME_DEOPT_SUFFIX);
+        }
+
+        @Override
+        public List<String> paramTypes() {
+            LinkedList<String> paramTypes = new LinkedList<>();
+            Signature signature = hostedMethod.getSignature();
+            for (int i = 0; i < signature.getParameterCount(false); i++) {
+                final JavaType parameterType = signature.getParameterType(i, null);
+                paramTypes.add(toJavaName(parameterType));
+            }
+            return paramTypes;
         }
 
         @Override
-        public int getModifiers() {
+        public List<String> paramNames() {
+            /* Can only provide blank names for now. */
```

Comment:
How about
```java
String[] paramNames = new String[signature.getParameterCount(false)] ;
Arrays.fill(paramNames, "");
return Arrays.asList(paramNames);
```

Reply:
Sure, but at some point `paramNames` will not be `""`.